num_workers makes the provider crash (workaround: use autoscale)

Hi there,



### Terraform Version
0.12.16

### Affected Resource(s)
databricks_cluster


### Terraform Configuration Files
```hcl
resource "databricks_cluster" "data-test" {
  cluster_name  = var.rg-name
  spark_version = var.databricks-version
  node_type_id  = var.databricks-node-size
  driver_node_type_id  = var.databricks-node-size
  num_workers   = var.databricks-num-workers

#  autoscale {
#    min_workers = 1
#    max_workers = 2
#  }

  spark_conf = {
    "spark.databricks.delta.preview.enabled" = true
  }
  autotermination_minutes = var.databricks-termination-delay
}
```

### Debug Output
```
databricks_cluster.data-test: Creating...

Error: rpc error: code = Unavailable desc = transport is closing


panic: interface conversion: interface {} is int, not int32
2020-04-14T17:31:17.650+0200 [DEBUG] plugin.terraform-provider-databricks:
2020-04-14T17:31:17.650+0200 [DEBUG] plugin.terraform-provider-databricks: goroutine 15 [running]:
2020-04-14T17:31:17.650+0200 [DEBUG] plugin.terraform-provider-databricks: github.com/innovationnorway/terraform-provider-databricks/databricks.resourceDatabricksClusterCreate(0xc0002e2540, 0xe50300, 0xc00073c000, 0x2, 0x19e9a80)
2020-04-14T17:31:17.650+0200 [DEBUG] plugin.terraform-provider-databricks:      /data/tmp/analytics/terraform-provider-databricks/databricks/resource_databricks_cluster.go:394 +0xf70
2020-04-14T17:31:17.650+0200 [DEBUG] plugin.terraform-provider-databricks: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc0003af7a0, 0xc0000c4be0, 0xc0005606a0, 0xe50300, 0xc00073c000, 0xc0003aa901, 0xc000526c30, 0xc0003aa980)
2020-04-14T17:31:17.650+0200 [DEBUG] plugin.terraform-provider-databricks:      /data/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.9.1/helper/schema/resource.go:310 +0x365
2020-04-14T17:31:17.651+0200 [DEBUG] plugin.terraform-provider-databricks: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc00017c900, 0xc0002839c8, 0xc0000c4be0, 0xc0005606a0, 0xc00051af28, 0xc0000107a0, 0xed5e00)
2020-04-14T17:31:17.651+0200 [DEBUG] plugin.terraform-provider-databricks:      /data/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.9.1/helper/schema/provider.go:294 +0x99
2020-04-14T17:31:17.651+0200 [DEBUG] plugin.terraform-provider-databricks: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc0001404a0, 0x12a1c40, 0xc000690270, 0xc0002e2000, 0xc0001404a0, 0xc000690270, 0xc00023bb30)
2020-04-14T17:31:17.651+0200 [DEBUG] plugin.terraform-provider-databricks:      /data/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.9.1/internal/helper/plugin/grpc_provider.go:885 +0x8b4
2020-04-14T17:31:17.651+0200 [DEBUG] plugin.terraform-provider-databricks: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x100c820, 0xc0001404a0, 0x12a1c40, 0xc000690270, 0xc0000924e0, 0x0, 0x12a1c40, 0xc000690270, 0xc0004fe700, 0x361)
2020-04-14T17:31:17.651+0200 [DEBUG] plugin.terraform-provider-databricks:      /data/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.9.1/internal/tfplugin5/tfplugin5.pb.go:3305 +0x217
2020-04-14T17:31:17.651+0200 [DEBUG] plugin.terraform-provider-databricks: google.golang.org/grpc.(*Server).processUnaryRPC(0xc00053cf00, 0x12adea0, 0xc00053d980, 0xc000162200, 0xc0000b3c20, 0x19bcf60, 0x0, 0x0, 0x0)
2020-04-14T17:31:17.651+0200 [DEBUG] plugin.terraform-provider-databricks:      /data/go/pkg/mod/google.golang.org/grpc@v1.27.1/server.go:1024 +0x4f4
2020-04-14T17:31:17.651+0200 [DEBUG] plugin.terraform-provider-databricks: google.golang.org/grpc.(*Server).handleStream(0xc00053cf00, 0x12adea0, 0xc00053d980, 0xc000162200, 0x0)
2020-04-14T17:31:17.651+0200 [DEBUG] plugin.terraform-provider-databricks:      /data/go/pkg/mod/google.golang.org/grpc@v1.27.1/server.go:1313 +0xd97
2020-04-14T17:31:17.651+0200 [DEBUG] plugin.terraform-provider-databricks: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0004d0110, 0xc00053cf00, 0x12adea0, 0xc00053d980, 0xc000162200)
2020-04-14T17:31:17.651+0200 [DEBUG] plugin.terraform-provider-databricks:      /data/go/pkg/mod/google.golang.org/grpc@v1.27.1/server.go:722 +0xbb
2020-04-14T17:31:17.651+0200 [DEBUG] plugin.terraform-provider-databricks: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-04-14T17:31:17.651+0200 [DEBUG] plugin.terraform-provider-databricks:      /data/go/pkg/mod/google.golang.org/grpc@v1.27.1/server.go:720 +0xa1
2020/04/14 17:31:17 [DEBUG] databricks_cluster.data-test: apply errored, but we're indicating that via the Error pointer rather than returning it: rpc error: code = Unavailable desc = transport is closing
2020/04/14 17:31:17 [TRACE] <root>: eval: *terraform.EvalMaybeTainted
2020/04/14 17:31:17 [TRACE] EvalMaybeTainted: databricks_cluster.data-test encountered an error during creation, so it is now marked as tainted
2020/04/14 17:31:17 [TRACE] <root>: eval: *terraform.EvalWriteState
2020/04/14 17:31:17 [TRACE] EvalWriteState: removing state object for databricks_cluster.data-test
2020/04/14 17:31:17 [TRACE] <root>: eval: *terraform.EvalApplyProvisioners
2020/04/14 17:31:17 [TRACE] EvalApplyProvisioners: databricks_cluster.data-test has no state, so skipping provisioners
2020/04/14 17:31:17 [TRACE] <root>: eval: *terraform.EvalMaybeTainted
2020/04/14 17:31:17 [TRACE] EvalMaybeTainted: databricks_cluster.data-test encountered an error during creation, so it is now marked as tainted
2020/04/14 17:31:17 [TRACE] <root>: eval: *terraform.EvalWriteState
2020/04/14 17:31:17 [TRACE] EvalWriteState: removing state object for databricks_cluster.data-test
2020/04/14 17:31:17 [TRACE] <root>: eval: *terraform.EvalIf
2020/04/14 17:31:17 [TRACE] <root>: eval: *terraform.EvalIf
2020/04/14 17:31:17 [TRACE] <root>: eval: *terraform.EvalWriteDiff
2020/04/14 17:31:17 [TRACE] <root>: eval: *terraform.EvalApplyPost
2020/04/14 17:31:17 [ERROR] <root>: eval: *terraform.EvalApplyPost, err: rpc error: code = Unavailable desc = transport is closing
2020/04/14 17:31:17 [ERROR] <root>: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020/04/14 17:31:17 [TRACE] [walkApply] Exiting eval tree: databricks_cluster.data-test
2020/04/14 17:31:17 [TRACE] vertex "databricks_cluster.data-test": visit complete
2020/04/14 17:31:17 [TRACE] dag/walk: upstream of "provider.databricks (close)" errored, so skipping
2020/04/14 17:31:17 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2020/04/14 17:31:17 [TRACE] dag/walk: upstream of "root" errored, so skipping
2020-04-14T17:31:17.652+0200 [DEBUG] plugin: plugin process exited: path="/mnt/c/Users/mvdborne/OneDrive - Isabel Group/Documents/.terraform.d/plugins/linux_amd64/terraform-provider-databricks" pid=5860 error="exit status 2"
2020-04-14T17:31:17.740+0200 [DEBUG] plugin: plugin exited



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

SECURITY WARNING: the "crash.log" file that was created may contain
sensitive information that must be redacted before it is safe to share
on the issue tracker.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

### Panic Output

### Expected Behavior
Cluster created.

### Actual Behavior
Terraform crashes when num_workers is set, but not when autoscale is set.

### Steps to Reproduce
terraform apply

--- a/databricks/resource_databricks_cluster.go
+++ b/databricks/resource_databricks_cluster.go
@@ -388,7 +388,7 @@ func resourceDatabricksClusterCreate(d *schema.ResourceData, meta interface{}) e
 	}
 
 	if v, ok := d.GetOk("num_workers"); ok {
-		attributes.NumWorkers = to.Int32Ptr(v.(int32))
+		attributes.NumWorkers = to.Int32Ptr(int32(v.(int)))
 	}
 
 	if v, ok := d.GetOk("autoscale"); ok {
@@ -507,7 +507,7 @@ func resourceDatabricksClusterUpdate(d *schema.ResourceData, meta interface{}) e
 	}
 
 	if v, ok := d.GetOk("num_workers"); ok {
-		attributes.NumWorkers = to.Int32Ptr(v.(int32))
+		attributes.NumWorkers = to.Int32Ptr(int32(v.(int)))
 	}
 
 	if v, ok := d.GetOk("autoscale"); ok {

--- a/databricks/resource_databricks_cluster_test.go
+++ b/databricks/resource_databricks_cluster_test.go
@@ -9,6 +9,47 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/terraform"
 )
 
+func TestAccDatabricksCluster_AutoScale(t *testing.T) {
+	resourceName := "databricks_cluster.test"
+	clusterName := acctest.RandString(6)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDatabricksClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatabricksClusterAutoScale(clusterName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "cluster_name", clusterName),
+					resource.TestCheckResourceAttr(resourceName, "autoscale.0.min_workers", "2"),
+					resource.TestCheckResourceAttr(resourceName, "autoscale.0.max_workers", "8"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDatabricksCluster_NumWorkers(t *testing.T) {
+	resourceName := "databricks_cluster.test"
+	clusterName := acctest.RandString(6)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDatabricksClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatabricksClusterNumWorkers(clusterName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "cluster_name", clusterName),
+					resource.TestCheckResourceAttr(resourceName, "num_workers", "2"),
+				),
+			},
+		},
+	})
+}
+
 func TestAccDatabricksCluster_Azure(t *testing.T) {
 	resourceName := "databricks_cluster.test"
 	clusterName := acctest.RandString(6)
@@ -74,6 +115,37 @@ func testAccCheckDatabricksClusterDestroy(s *terraform.State) error {
 	}
 
 	return nil
+}
+
+func testAccDatabricksClusterAutoScale(clusterName string) string {
+	return fmt.Sprintf(`
+resource "databricks_cluster" "test" {
+  cluster_name  = "%s"
+  spark_version = "6.3.x-scala2.11"
+  node_type_id  = "Standard_DS3_v2"
+
+  autoscale {
+    min_workers = 2
+    max_workers = 8
+  }
+
+  autotermination_minutes = 120
+}
+`, clusterName)
+}
+
+func testAccDatabricksClusterNumWorkers(clusterName string) string {
+	return fmt.Sprintf(`
+resource "databricks_cluster" "test" {
+  cluster_name  = "%s"
+  spark_version = "6.3.x-scala2.11"
+  node_type_id  = "Standard_DS3_v2"
+
+  num_workers = 2
+
+  autotermination_minutes = 120
+}
+`, clusterName)
 }
 
 func testAccDatabricksClusterAzure(clusterName string) string {

